app/models/upload.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+gem 'activerecord-import'
+
+class Upload
+  include ActiveModel::Validations
+
+  attr_reader :search_engine, :csv_file
+
+  validates_with UploadValidator
+
+  def initialize(search_engine:, csv_file:)
+    @search_engine = search_engine
+    @csv_file = csv_file
+  end
+
+  def save
+    return false unless valid?
+
+    begin
+      content = csv_file.read
+      keywords = CSV.parse(content).flatten.map(&:strip)
+      search_results = keywords.map { |keyword| SearchResult.new(keyword: keyword, search_engine: search_engine) }
+      SearchResult.import search_results, validate: true, raise_error: true
+    rescue StandardError
```

Comment:
Is there not a specific custom error thrown instead of catching with `StandardError`? The more specific, the better,